Compilation error on assigning to a tuple.

```
## pythran export test_LoopMaster()
import numpy as np

def LooperMaster___init__():
    self_userUseTempo = 1
    self = tuple([self_userUseTempo])
    return self

def test_LoopMaster():
    ML = LooperMaster___init__()
    ML[0] = 1
    print('LoopMaster OK')
```
This used to work in previous version of pythran. No longer works. 
```
CRITICAL: I am in trouble. Your input file does not seem to match Pythran's constraints...
TD1.py:11:4 error: Assign to unknown node
----
    ML[0] = 1
    ^~~~ (o_0)
----
```

--- a/pythran/analyses/constant_expressions.py
+++ b/pythran/analyses/constant_expressions.py
@@ -49,6 +49,7 @@ class ConstantExpressions(NodeAnalysis):
 
     def visit_Subscript(self, node):
         rec = all([self.visit(x) for x in (node.value, node.slice)])
+        rec = isinstance(node.ctx, ast.Load) and rec
         return rec and self.add(node)
 
     def visit_Name(self, node):

--- a/pythran/optimizations/range_based_simplify.py
+++ b/pythran/optimizations/range_based_simplify.py
@@ -84,6 +84,6 @@ class RangeBasedSimplify(Transformation):
     def visit_Name(self, node):
         if isinstance(node.ctx, ast.Load):
             return self.visit_range(node)
-        return node
+        return self.generic_visit(node)
 
-    visit_Subscript = visit_range
+    visit_Subscript = visit_Name

--- a/pythran/tests/test_optimizations.py
+++ b/pythran/tests/test_optimizations.py
@@ -667,3 +667,15 @@ class TestAnalyses(TestEnv):
                 return data'''
         self.run_test(code, 'aa', 2, 'bb', '3', subscript_function_aliasing=[str, int, str, str])
 
+    def test_range_simplify_subscript(self):
+        code = '''
+def LooperMaster___init__():
+    self_userUseTempo = 1
+    self = [self_userUseTempo]
+    return self
+
+def range_simplify_subscript(n):
+    ML = LooperMaster___init__()
+    ML[0] = n
+    return ML'''
+        self.run_test(code, 1, range_simplify_subscript=[int])

